# Clase 9.1 Regression simple, multiple,...

In [1]:
%ls datasets/

adult.csv                  census.csv     movies.csv
airline_data.csv           coches.csv     nhanes_2015_2016.csv
antropometria00.csv        cwurData.csv   pokemon.csv
antropometria-dataset.csv  data.csv       reg_data.csv
Book3.csv                  headbrain.csv  sample_pivot.csv
Canada.xlsx                mic2.xlsx      tips.csv
Cartwheeldata.csv          mic.xlsx       weight-height.csv


In [2]:
import pandas as pd

In [3]:
df = pd.read_csv('datasets/reg_data.csv')
df

Gender  Salary  Age       Place  Weight Company Academic_degree
0   Female    1500   33     Chicago    80.0     BMW        Bachelor
1   Female    1200   33     Chicago    82.5    Ford              No
2     Male    2200   34    New York   100.8     BMW        Bachelor
3     Male    2100   42    New York    90.0     BMW          Master
4   Female    1500   29     Chicago    67.0    Ford          Master
5   Female    1700   42  Washington    60.0    Ford          Master
6     Male    3000   50  Washington    77.0    Ford              No
7     Male    3000   55  Washington    77.0    Ford        Bachelor
8   Female    2800   31    New York    87.0    Ford        Bachelor
9     Male    2900   46    New York    70.0      GM          Master
10  Female    2780   36  Washington    57.0     BMW              No
11    Male    2550   48    New York    64.0      GM          Master

In [4]:
df.columns

Index(['Gender', 'Salary', 'Age', 'Place', 'Weight', 'Company',
       'Academic_degree'],
      dtype='object')

# Introduccion
Esta es una base de datos que explica el salario de 12 personas desde diferentes empresas
## mis variables de interes
* 'Gender',: Explica el genero de los empleados, es una variable categorica, tiene 2 categorias
* 'Salary',;... 
* 'Age', 
* 'Academic_degree'],


# 1. Pregunta de investigacion 1: 
### Cual es el impacto de Age en Salary

Salary = b0+ b1*Age

In [5]:
#pip install statsmodels
#pip install seaborn
#pip install sklearn

In [6]:
import matplotlib.pyplot as plt
import seaborn as sns
import statsmodels.api as sm
import numpy as nms
from sklearn import linear_model

In [7]:
modelA = sm.OLS.from_formula('Salary~Age', data = df)
resultA = modelA.fit()
resultA.summary()

/srv/conda/envs/notebook/lib/python3.7/site-packages/scipy/stats/stats.py:1535: UserWarning: kurtosistest only valid for n>=20 ... continuing anyway, n=12
  "anyway, n=%i" % int(n))


<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                 Salary   R-squared:                       0.380
Model:                            OLS   Adj. R-squared:                  0.319
Method:                 Least Squares   F-statistic:                     6.142
Date:                Tue, 30 Nov 2021   Prob (F-statistic):             0.0326
Time:                        13:42:00   Log-Likelihood:                -91.514
No. Observations:                  12   AIC:                             187.0
Df Residuals:                      10   BIC:                             188.0
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept    349.4458    790.373      0.442      0.668   -1411.614    2110.506
Age           48.0932     19.406      2.478      0.033       4.853      91.333
==============================================================================
Omnibus:                        0.406   Durbin-Watson:                   1.484
Prob(Omnibus):                  0.816   Jarque-Bera (JB):                0.479
Skew:                           0.321   Prob(JB):                        0.787
Kurtosis:                       2.261   Cond. No.                         205.
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

### Observaciones: 
* Variable dependiente: Salary
* Variable Independiente: Age 
* R-squared: 0.380; por lo tanto, este modelo no es tan efficiente. Porque explica solo 38% salary basado en age.
* Df Model: 1
* intercept ... >0.05, no es estadisticamente significativo

Salary = 349.4458 + 48.0932*Age

In [8]:
print('Salary of a person whose age is 33: \t', 349.4458 + 48.0932*33)
df.iloc[:2, :]

Salary of a person whose age is 33: 	 1936.5214


Gender  Salary  Age    Place  Weight Company Academic_degree
0  Female    1500   33  Chicago    80.0     BMW        Bachelor
1  Female    1200   33  Chicago    82.5    Ford              No

In [12]:
print('Diferencia es: \t:', 1500- 1936.5214)
print('Diferencia es: \t:', 1200- 1936.5214)
resultA.resid

Diferencia es: 	: -436.5214000000001
Diferencia es: 	: -736.5214000000001


0    -436.521924
1    -736.521924
2     215.384860
3    -269.360866
4    -244.149060
5    -669.360866
6     245.893407
7       5.427328
8     959.664508
9     338.266270
10    699.198429
11   -107.920161
dtype: float64